src/index.js

```diff
@@ -104,22 +108,55 @@ HTTPSnippet.prototype.prepare = function (request) {
       if (request.postData.params) {
         var form = new MultiPartForm()
 
+        // The `form-data` module returns one of two things: a native FormData object, or its own polyfill. Since the
+        // polyfill does not support the full API of the native FormData object, when this library is running in a
+        // browser environment it'll fail on two things:
+        //
+        //  - The API for `form.append()` has three arguments and the third should only be present when the second is a
+        //    Blob or USVString.
+        //  - `FormData.pipe()` isn't a function.
+        //
+        // Since the native FormData object is iterable, we easily detect what version of `form-data` we're working
+        // with here to allow `multipart/form-data` requests to be compiled under both browser and Node environments.
+        //
+        // This hack is pretty awful but it's the only way we can use this library in the browser as if we code this
+        // against just the native FormData object, we can't polyfill that back into Node because Blob and File objects,
+        // which something like `formdata-polyfill` requires, don't exist there.
+        const isNativeFormData = (typeof form[Symbol.iterator] === 'function')
```

Comment:
It's unfortunate that the check needs to be this way, but awesome explanation nonetheless 💯

Reply:
Just an idea, but another option may be to require the polyfill directly from the module, and do an `instanceof` check? Untested code follows 😄 

```js
var MultiPartForm = require('form-data');
var FormDataPolyfil = require('form-data/lib/form_data');

var form = new MultiPartForm();
const isNativeFormData = !(form instanceof FormDataPolyfil);
```

I noticed they do the same check in the [form-data source](https://github.com/form-data/form-data/blob/55d90ce4a4c22b0ea0647991d85cb946dfb7395b/lib/form_data.js#L27), although to confirm initialization.

Reply:
Good idea.